Feature/hook use hover

seregga opened the pull request.

seregga committed:
feat: developed react hook useFetch

Task: 1 in react_hooks

seregga committed:
fix: deleted import hook useRef

seregga committed:
chore: removed commented code

Task: 1 in react_hooks

seregga committed:
feat: developed react hook useLocalStorage

Task: 2 in react_hooks

seregga committed:
feat: developed react hook useHover

Task: 3 in react_hooks

antisedative commented on src/hooks/useHover.js:
```diff
@@ -0,0 +1,19 @@
+        elem.addEventListener('mouseenter', () => setHovered(true))
+        elem.addEventListener('mouseleave', () => setHovered(false))
+
+        return () => {
+            elem.removeEventListener('mouseenter', () => setHovered(true))
+            elem.removeEventListener('mouseleave', () => setHovered(false))
+        }
```
обработчики не будут удаляться, тк передаешь анонимные функции
функции должны совпадать по ссылке

seregga replied:
исправлено

antisedative commented on src/hooks/useHover.js:
```diff
@@ -0,0 +1,19 @@
+    const ref = useRef(null)
+    
+    useEffect(() => {
+        const elem = ref.current
+        elem.addEventListener('mouseenter', () => setHovered(true))
+        elem.addEventListener('mouseleave', () => setHovered(false))
```
перед добавлением слушателей лучше проверить существует ли elem, иначе упадет ошибка

seregga replied:
исправлено

seregga committed:
fix: hook useHover

1. Added a check for the element
2. Functions are placed in variables for the
correct removal of event handlers.

Task: 3 in react_hooks
